x-pack/plugins/cloud_security_posture/common/utils/helpers.ts

```diff
@@ -25,13 +36,30 @@ export const extractErrorMessage = (e: unknown, defaultMessage = 'Unknown Error'
   return defaultMessage; // TODO: i18n
 };
 
-export const createCspRuleSearchFilterByPackagePolicy = ({
-  packagePolicyId,
-  policyId,
-}: {
-  packagePolicyId: string;
-  policyId?: string;
-}): string =>
-  `${CSP_RULE_SAVED_OBJECT_TYPE}.attributes.package_policy_id: "${packagePolicyId}"${
-    policyId ? ` AND ${CSP_RULE_SAVED_OBJECT_TYPE}.attributes.policy_id: "${policyId}"` : ''
-  }`;
+export const getBenchmarkTypeFilterFromBenchmarkId = (type: BenchmarkId): string =>
+  `${CSP_RULE_TEMPLATE_SAVED_OBJECT_TYPE}.attributes.metadata.benchmark.id: "${type}"`;
+
+export const isEnabledBenchmarkInputType = (input: PackagePolicyInput | NewPackagePolicyInput) =>
+  !!input.type && input.enabled;
+
+export const isCspPackage = (packageName?: string) =>
+  packageName === CLOUD_SECURITY_POSTURE_PACKAGE_NAME;
+
+export const extractBenchmarkFromPackagePolicy = (
+  inputs: PackagePolicy['inputs'] | NewPackagePolicy['inputs']
+): BenchmarkId => {
+  const enabledInputs = inputs.filter(isEnabledBenchmarkInputType);
+
+  // Use the only enabled input
+  if (enabledInputs.length === 1) {
+    return getInputType(enabledInputs[0].type);
+  }
+
+  // Use the default benchmark id for multiple/none selected
+  return getInputType(CLOUDBEAT_VANILLA);
+};
```

Comment:
This is a function that was created already before, I just moved it here to this `helpers.ts` file.

I don't mind changing it, and the things you said make sense, but I wonder if there is any scenario that we are missing here. I am a bit concerned it would be a breaking change.

WDYT?

Reply:
i think it was wrong to begin with (i think i added this?) 😄 

say you're on EKS, and for some reason both inputs are enabled, we'd select the `cis_k8s` as the default and the rules page UI or the benchmark API would both return the wrong results. 

i believe this didn't happen as we ensure only a single input is enabled when installing. 

with the addition of `CSPM`, this just becomes even more likely to be a bug as we have more chances to be wrong about the default type we provide. 

in short, i'm not sure if it should be done in this PR or another one, but given that we only support a single enabled input per policy, i'm pretty sure we should throw here

Reply:
If that's Ok, I prefer to split it into a different PR